src/safeposix/syscalls/fs_calls.rs

```diff
@@ -2261,107 +2261,198 @@ impl Cage {
+            //sure that chmod_syscall() is only called on the supported file types.
+            //BUG: S_IFIFO is included in S_FILETYPEFLAGS implying that calling chmod_syscall() on named papes is supported even though named pipes
+            //are currently not supported. 
             if mode & (S_IRWXA | (S_FILETYPEFLAGS as u32)) == mode {
                 Self::_chmod_helper(inodenum, mode);
             } else {
                 //there doesn't seem to be a good syscall error errno for this
-                return syscall_error(Errno::EACCES, "chmod", "provided file mode is not valid");
+                return syscall_error(Errno::EINVAL, "chmod", "The value of the mode argument is invalid");
             }
         } else {
-            return syscall_error(Errno::ENOENT, "chmod", "the provided path does not exist");
+            return syscall_error(Errno::ENOENT, "chmod", "A component of path does not name an existing file");
         }
         0 //success!
     }
 
-    //------------------------------------FCHMOD SYSCALL------------------------------------
+    /// ### Description
+    ///
+    /// The `fchmod_syscall()` is equivalent to chmod() in that it is used to change a file's mode bits 
+    /// that consist of read, write, and execute file permission bits except that the file
+    /// is specified by the file descriptor.
+    /// Changing set-user-ID, set-group-ID, and sticky bits is currently not supported.
+    ///
+    /// ### Arguments
+    ///
+    /// The `fchmod_syscall()` accepts two arguments:
+    /// * `fd` - an open file descriptor.
+    /// * `mode` - the new file mode, which is a bit mask created by bitwise-or'ing zero or more valid mode bits.
+    /// Some of the examples of such bits are `S_IRUSR` (read by owner), `S_IWUSR` (write by owner), 
+    /// `S_ISUID` (set-user-ID), etc.
+    ///
+    /// ### Returns
+    /// 
+    /// Upon successful completion, zero is returned.
+    /// In case of a failure, -1 is returned, and `errno` is set depending on the error, e.g. EACCES, ENOENT, etc.
+    ///
+    /// ### Errors and Panics
+    ///
+    /// * `EBADF` - the file descriptor fd is not valid. 
+    /// * `EINVAL` - the value of the `mode` argument is invalid or mode bits cannot be changed on this file type
+    /// Other errors, like `EFAULT  , `ENOTDIR`, etc. are not supported.
+    ///
+    /// A panic occurs when a provided file descriptor is out of bounds
+    ///
+    /// To learn more about the syscall, valid mode bits, and error values, see
+    /// [fchmod(2)](https://linux.die.net/man/2/fchmod)
 
     pub fn fchmod_syscall(&self, fd: i32, mode: u32) -> i32 {
+        //BUG
+        //if the provided file descriptor is out of bounds, 'get_filedescriptor' returns Err(),
+        //unwrapping on which  produces a 'panic!'
+        //otherwise, file descriptor table entry is stored in 'checkedfd'
         let checkedfd = self.get_filedescriptor(fd).unwrap();
```

Comment:
Should we return an error or panic in this case?

Reply:
This is a bug that we have discussed with Nick. Currently, if the descriptor is out of bounds, it will panic. For now, we just marked it as a bug.